src/domain/event.ts

```diff
@@ -0,0 +1,125 @@
+import { Address, BigInt, ByteArray, Bytes, crypto } from '@graphprotocol/graph-ts';
+import { Event, Proposal, ReputationHolder } from '../types/schema';
+import { concat } from '../utils';
+
+export function addNewDAOEvent(avatar: Address, daoName: string, timestamp: BigInt): void {
+    addEvent(
+        'NewDAO',
+        avatar.toHex(),
+        '{ "address": "' + avatar.toHex() + '", "name": "' + daoName + '" }',
```

Comment:
what is that ?
why it is hard coded here?

Reply:
That’s just the data object Alchemy needs for this event type...

Reply:
on first glance it looks not so elegant solution. is that can be avoided ? @jellegerbrandy

Reply:
tibet is right: the avatar is not strictly needed (although neither does it hurt to have it :-))

I do not really understand @orenyodfat's objection. 
Events have a `data` attribute that is a string. This string happends to be a JSON structure, which is useful when we want to reprseent the event. This does not seem particularly unelegant to me as a pattern

Reply:
Does this data need to be searchable and indexed?
It looks too specific and duplicate the field we  have on the entity it self .

Reply:
no it does not, that is why it is in a text field and not in its own separate field.